### PR TITLE
Fix obscure error messages

### PR DESCRIPTION
**Pull request by sougou**
Two fixes:
* `gtid_mode is OFF` was a misleading error message. It's been changed to `replication may not be running`.
* Topo errors were too obscure because they did not specify the node on which they were failing. The errors now carry a type code, which allows us to add more information to the error message.

### CODE REVIEW

**Comment by sougou** [comment]
The core of the change:
* Hardcoded errors have been removed from `go/vt/topo/server.go`.
* They have been replaced by new errors in `go/vt/topo/errors.go`.

The rest are the fallout of these two changes.

File: go/vt/topo/errors.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	case Interrupted:
+		message = fmt.Sprintf("interrupted: %s", node)
+	case BadVersion:
+		message = fmt.Sprintf("bad node version: %s", node)
+	case PartialResult:
+		message = fmt.Sprintf("partial result: %s", node)
+	case NoUpdateNeeded:
+		message = fmt.Sprintf("no update needed: %s", node)
+	default:
+		message = fmt.Sprintf("unknonw code: %s", node)
```
Comment (demmer) [inline go/vt/topo/errors.go]: typo: "unknown"

File: go/vt/wrangler/schema.go [hunk 2]
```diff
@@ -98,7 +98,12 @@ func (wr *Wrangler) ReloadSchemaShard(ctx context.Context, keyspace, shard, repl
 			defer wg.Done()
 			concurrency.Acquire()
 			defer concurrency.Release()
-			if err := wr.tmc.ReloadSchema(ctx, tablet, replicationPos); err != nil {
+			pos := replicationPos
+			// Master is always up-to-date. So, don't wait for position.
+			if tablet.Type == topodatapb.TabletType_MASTER {
+				pos = ""
+			}
+			if err := wr.tmc.ReloadSchema(ctx, tablet, pos); err != nil {
```
Comment (demmer) [inline go/vt/wrangler/schema.go]: This seems like an unrelated change to the PR description -- I _assume_ this is a separate bug which triggered the whole sweeping error reporting change, but if so it should either be called out explicitly in the PR or (better IMO) pulled out into a separate bug fix change and tested.

Comment (sougou) [reply go/vt/wrangler/schema.go]: The only change in behavior is that the annoying warnings don't get printed, which I tested manually. As for the PR, the description calls out this change, and it's a separate commit. But I can pull this out in a separate PR if you insist.

I'll address the other comments momentarily.

**Comment by alainjobart** [comment]
Just saw this change as I am importing it into our code base. This is awesome, long overdue. +10
